Assets/Scripts/Controllers/ObstacleController.cs

```diff
@@ -1,23 +1,30 @@
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
+using UnityEngine.Tilemaps;
 
 public class ObstacleController : MonoBehaviour
 {
-    
-    [SerializeField] private float obstacleSpeed;
-    private Vector3 offset = new Vector3(0, 0.25f);
-    private float timer;
+    [SerializeField] private GameObject[] obstacleTypes;
 
-    // Start is called before the first frame update
-    void Start()
+    public void SpawnObstacle(GameObject parent)
     {
+        var obstacleType = GetRandomObstacleType();
+        var obstacle = Instantiate(obstacleType, parent.transform);
+
+        var location = GetRandomOffset(parent);
+        obstacle.transform.position += location;
+    }
+
+    private GameObject GetRandomObstacleType()
+    {
+        return obstacleTypes[Random.Range(0, obstacleTypes.Length)];
     }
 
-    // Update is called once per frame
-    void Update()
+    private Vector3 GetRandomOffset(GameObject parentChunk)
     {
-        timer += Time.fixedDeltaTime;
-        transform.localPosition = new Vector3(Mathf.Sin(timer), Mathf.Sin(timer) / 2) * obstacleSpeed * Time.fixedDeltaTime + offset;
+        var chunk = parentChunk.GetComponent<Chunk>();
+        return chunk.GetRandomTileOffset();
```

Comment:
Using a method exposed on the script that I probably shouldn't have access to...